driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -97,4 +97,24 @@ public interface Expression {
      * @return true if less than or equal to, false otherwise
      */
     BooleanExpression lte(Expression lte);
+
+    /**
+     * also checks for nulls
+     * @param or
+     * @return
+     */
+    BooleanExpression isBooleanOr(BooleanExpression or);
+    NumberExpression isNumberOr(NumberExpression or);
```

Comment:
The provided API seems to be providing access to a subset of the use cases for type tests. Can you clarify why we don't need:

```
BooleanExpression isNumber()
```

so that a type test can be used as a condition in arbitrary conditional expressions?

Reply:
Discussion here: https://github.com/mongodb/mongo-java-driver/pull/1050#discussion_r1036374941